### PR TITLE
Unify logging - log4j2

### PR DESCRIPTION
**Pull request by Beacze**
- replace log4j with log4j2
- remove slf4j
- remove org.jboss.logging
- fix logging in all modules
- comment poi and ugh libraries - they need log4j

### CODE REVIEW

File: Kitodo-DataManagement/src/main/java/org/kitodo/config/ConfigMain.java [hunk 1]
```diff
@@ -14,11 +14,11 @@
...
 import org.apache.commons.configuration.PropertiesConfiguration;
 import org.apache.commons.configuration.reloading.FileChangedReloadingStrategy;
-import org.apache.log4j.Level;
-import org.apache.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 public class ConfigMain {
-    private static final Logger myLogger = Logger.getLogger(ConfigMain.class);
+    private static final Logger myLogger = LogManager.getLogger(ConfigMain.class);
```
Comment (Kathrin-Huber) [inline Kitodo-DataManagement/src/main/java/org/kitodo/config/ConfigMain.java]: myLogger

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/database/helper/Helper.java [hunk 2]
```diff
@@ -53,7 +54,7 @@ private void writeObject(ObjectOutputStream aOutputStream) throws IOException {
 
     }
 
-    private static final Logger myLogger = Logger.getLogger(Helper.class);
+    private static final Logger myLogger = LogManager.getLogger(Helper.class);
```
Comment (Kathrin-Huber) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/database/helper/Helper.java]: myLogger

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/HibernateSessionLong.java [hunk 3]
```diff
@@ -21,7 +22,7 @@
 // TODO: Fix for Hibernate-Session-Management, old Version reactivated
 
 public class HibernateSessionLong {
-    private static final Logger mylogger = Logger.getLogger(HibernateSessionLong.class);
+    private static final Logger mylogger = LogManager.getLogger(HibernateSessionLong.class);
```
Comment (Kathrin-Huber) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/HibernateSessionLong.java]: myLogger

File: Kitodo-FileManagement/src/main/java/org/kitodo/config/Config.java [hunk 4]
```diff
@@ -14,11 +14,11 @@
...
 import org.apache.commons.configuration.PropertiesConfiguration;
 import org.apache.commons.configuration.reloading.FileChangedReloadingStrategy;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 public class Config {
-    private static final Logger myLogger = LoggerFactory.getLogger(Config.class);
+    private static final Logger myLogger = LogManager.getLogger(Config.class);
```
Comment (Kathrin-Huber) [inline Kitodo-FileManagement/src/main/java/org/kitodo/config/Config.java]: myLogger

File: Kitodo-DataManagement/src/main/java/org/kitodo/config/ConfigMain.java (hunk 1 above)
Comment (Beacze) [inline Kitodo-DataManagement/src/main/java/org/kitodo/config/ConfigMain.java]: all fixed

File: Kitodo/src/main/java/org/kitodo/production/plugin/importer/massimport/googlecode/fascinator/redbox/sru/NLAIdentity.java [hunk 5]
```diff
@@ -40,7 +40,7 @@
  */
 public class NLAIdentity {
     /** Logging. **/
-    private static Logger log = LoggerFactory.getLogger(NLAIdentity.class);
+    private static Logger log = LogManager.getLogger(NLAIdentity.class);
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/production/plugin/importer/massimport/googlecode/fascinator/redbox/sru/NLAIdentity.java]: Should not `log` renamed to `logger`?

File: Kitodo/src/main/java/org/kitodo/production/plugin/importer/massimport/googlecode/fascinator/redbox/sru/SRUClient.java [hunk 6]
```diff
@@ -67,7 +67,7 @@
  */
 public class SRUClient {
     /** Logging. **/
-    private static Logger log = LoggerFactory.getLogger(SRUClient.class);
+    private static Logger log = LogManager.getLogger(SRUClient.class);
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/production/plugin/importer/massimport/googlecode/fascinator/redbox/sru/SRUClient.java]: Should not `log` renamed to `logger`?

File: Kitodo/src/main/java/org/kitodo/production/plugin/importer/massimport/googlecode/fascinator/redbox/sru/SRUResponse.java [hunk 7]
```diff
@@ -37,7 +37,7 @@
  */
 public class SRUResponse {
     /** Logging **/
-    private static Logger log = LoggerFactory.getLogger(SRUResponse.class);
+    private static Logger log = LogManager.getLogger(SRUResponse.class);
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/production/plugin/importer/massimport/googlecode/fascinator/redbox/sru/SRUResponse.java]: Should not `log` renamed to `logger`?

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/User.java [hunk 8]
```diff
@@ -321,7 +322,7 @@ public User selfDestruct() {
     // Here will be methods which should be in UserService but are used by jsp
     // files
 
-    private static final Logger logger = Logger.getLogger(MySQLHelper.class);
+    private static final Logger logger = LogManager.getLogger(MySQLHelper.class);
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/User.java]: Can you please correct the used class name from `MySQLHelper.class` to `User.class` as the class name is `User`?

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/apache/ProcessManager.java [hunk 9]
```diff
@@ -15,12 +15,13 @@
...
 
-import org.apache.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.kitodo.data.database.beans.Ruleset;
 
 public class ProcessManager {
 
-    private static final Logger logger = Logger.getLogger(MySQLHelper.class);
+    private static final Logger logger = LogManager.getLogger(MySQLHelper.class);
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/apache/ProcessManager.java]: Can you please correct the used class name from `MySQLHelper.class` to `ProcessManager.class` as the class name is `ProcessManager`?

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/apache/ProjectManager.java [hunk 10]
```diff
@@ -14,11 +14,12 @@
...
 import java.util.List;
 
-import org.apache.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.kitodo.data.database.beans.ProjectFileGroup;
 
 public class ProjectManager {
-    private static final Logger logger = Logger.getLogger(MySQLHelper.class);
+    private static final Logger logger = LogManager.getLogger(MySQLHelper.class);
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/apache/ProjectManager.java]: Can you please correct the used class name from `MySQLHelper.class` to `ProjectManager.class` as the class name is `ProjectManager`?

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/apache/SqlConfiguration.java [hunk 11]
```diff
@@ -29,7 +30,7 @@
     private String dbUser = "goobi";
     private String dbPassword = "CHANGEIT";
     private String dbURI = "jdbc:mysql://localhost/goobi?autoReconnect=true&amp;autoReconnectForPools=true";
-    private static final Logger logger = Logger.getLogger(MySQLHelper.class);
+    private static final Logger logger = LogManager.getLogger(MySQLHelper.class);
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/apache/SqlConfiguration.java]: Can you please correct the used class name from `MySQLHelper.class` to `SqlConfiguration.class` as the class name is `SqlConfiguration`?

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/apache/StepManager.java [hunk 12]
```diff
@@ -18,11 +18,12 @@
...
 import java.util.Map;
 
-import org.apache.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 public class StepManager {
 
-    private static final Logger logger = Logger.getLogger(MySQLHelper.class);
+    private static final Logger logger = LogManager.getLogger(MySQLHelper.class);
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/apache/StepManager.java]: Can you please correct the used class name from `MySQLHelper.class` to `StepManager.class` as the class name is `StepManager`?

File: Kitodo/src/main/java/org/goobi/mq/WebServiceResult.java [hunk 13]
```diff
@@ -15,14 +15,15 @@
...
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.joda.time.DateTime;
 import org.joda.time.format.DateTimeFormatter;
 import org.joda.time.format.ISODateTimeFormat;
 
 public class WebServiceResult {
-    private static final Logger logger = Logger.getLogger(ActiveMQDirector.class);
+    private static final Logger logger = LogManager.getLogger(ActiveMQDirector.class);
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/mq/WebServiceResult.java]: Can you please correct the used class name from `ActiveMQDirectory.class` to `WebServiceResult.class` as the class name is `WebServiceResult`?

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 14]
```diff
@@ -68,7 +69,7 @@
 
 public class CopyProcess extends ProzesskopieForm {
 
-    private static final Logger myLogger = Logger.getLogger(ProzesskopieForm.class);
+    private static final Logger logger = LogManager.getLogger(ProzesskopieForm.class);
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Can you please correct the used class name from `Prozesskopieform.class` to `CopyProcess.class` as the class name is `CopyProcess`?

File: Kitodo/src/main/java/de/sub/goobi/metadaten/copier/DataCopier.java [hunk 15]
```diff
@@ -26,7 +27,7 @@
  */
 public class DataCopier {
 
-    private static final Logger LOG = Logger.getLogger(DataCopier.class);
+    private static final Logger LOG = LogManager.getLogger(DataCopier.class);
```
Comment (Beacze) [inline Kitodo/src/main/java/de/sub/goobi/metadaten/copier/DataCopier.java]: all fixed

**Pull request merged**